client/src/components/deployButton.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import ButtonGroup from '@mui/material/ButtonGroup';
+import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
+import ClickAwayListener from '@mui/material/ClickAwayListener';
+import Grow from '@mui/material/Grow';
+import Paper from '@mui/material/Paper';
+import Popper from '@mui/material/Popper';
+import MenuItem from '@mui/material/MenuItem';
+import MenuList from '@mui/material/MenuList';
+
+const options = ['Deploy', 'Push to Hub and Deploy'];//, 'Push to OpenShift and Deploy'];
+
+interface DeployButtonProps {
+  onDeployClick: (mode: number) => void;
+  disabled?: boolean;
+}
+
+export default function DeployButton(props: DeployButtonProps) {
+  const onDeployClick = props.onDeployClick;
+  const disabled = props.disabled;
+  const [open, setOpen] = React.useState(false);
+  const anchorRef = React.useRef<HTMLDivElement>(null);
+  const [selectedIndex, setSelectedIndex] = React.useState(0);
+
+  const handleClick = () => {
+    onDeployClick(selectedIndex);
+  };
+
+  const handleMenuItemClick = (
+    _event: React.MouseEvent<HTMLLIElement, MouseEvent>,
+    index: number,
+  ) => {
+    setSelectedIndex(index);
+    setOpen(false);
+  };
+
+  const handleToggle = () => {
+    setOpen((prevOpen) => !prevOpen);
+  };
+
+  const handleClose = (event: Event) => {
+    if (
```

Comment:
``` javascript
setOpen( !!anchorRef.current && anchorRef.current.contains(event.target as HTMLElement))
```
or

``` javascript
setOpen(anchorRef.current !== null && anchorRef.current.contains(event.target as HTMLElement))
```
wold look better

Reply:
no it's a different logic. basically if (condition is met) then don't change the open state, else set open state to false.

Anyways, this code is coming from https://mui.com/material-ui/react-button-group/#split-button, so I trust it more than anything I can write ;-)